hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -182,6 +184,11 @@ public void notifyException(Throwable exception) {
             return;
         }
 
+        if (address != null && exception instanceof TargetNotMemberException) {
```

Comment:
My thinking was as following:
TargetNotMemberException is thrown when `address` is not in memberlist. It can be thrown from either server or client. 
If invocation intended to be send to an address `address != null`  it will retry always in same address. Address is not in memberlist, there is no point retrying. It does not matter if client message is retyable or not it will never be executed. 
Makes sense ? 

Note: `IMap.get` does not use `invokeOnTarget`,  it uses `invokeOnPartition`